### ISSUE TITLE
"Unhandled rejection Error: invalid operator" throwing in query/index.js:46:22

### ISSUE THREAD

**Issue by agonbina** [a8m/doqmentdb#28]
I'm trying out doqmentdb but I see this error being thrown all the time. I was testing if it can handle sending 50+ requests to DocumentDB every 500ms using setInterval, but this error is thrown multiple times, even though it seems like most of the operations complete successfully. Even if I try inserting one record every 500ms, this error is thrown at least once(but then again it seems all requests complete successfully). 

Here is all the code that I'm running: 

```js
var Doqment = require('doqmentdb')
var host = 'xxxxx'
var masterKey = 'xxxxx'
var connection = new (require('documentdb').DocumentClient)(host, {masterKey: masterKey})

var db = new Doqment(connection, 'test')
var users = db.use('users')

/**
* Model definition
*/

var model = {
	name: {
		type: String,
		default: ''
	}
}

users.schema(model)

users.pre('save', function(next) {
	this.createdAt = new Date().toString()
	next()
})

var errors = 0, successes = 0

function createUser(index) {
	users
		.create({ name: 'agon' })
		.then(function(data) {
			console.log('Created user:', data.id)
			successes += 1
		})
		.catch(function() {
			console.log('Error at i=%d', index)
			errors += 1
		})
}

var createUsers = setInterval(function() {
	for(var i = 0; i < 50; i += 1) {
		createUser(i)
	}
}, 500)

setTimeout(function() {
	clearInterval(createUsers)
	console.log('Success: %d', successes)
	console.log('Errors: %d', errors)
}, 5000)
```
And this is the trace log of the error: 
```js
Unhandled rejection Error: invalid operator
    at Error (<anonymous>)
    at /home/agonbina/Playground/doqmentdb-test/node_modules/doqmentdb/lib/query/index.js:46:22
    at Array.map (native)
    at query (/home/agonbina/Playground/doqmentdb-test/node_modules/doqmentdb/lib/query/index.js:26:27)
    at module.exports (/home/agonbina/Playground/doqmentdb-test/node_modules/doqmentdb/lib/query/index.js:80:36)
    at Manager.findOrCreate (/home/agonbina/Playground/doqmentdb-test/node_modules/doqmentdb/lib/manager.js:40:17)
    at /home/agonbina/Playground/doqmentdb-test/node_modules/doqmentdb/lib/collection.js:428:15
    at /home/agonbina/Playground/doqmentdb-test/node_modules/doqmentdb/node_modules/bluebird/js/main/using.js:165:30
    at tryCatcher (/home/agonbina/Playground/doqmentdb-test/node_modules/doqmentdb/node_modules/bluebird/js/main/util.js:24:31)
    at Promise._settlePromiseFromHandler (/home/agonbina/Playground/doqmentdb-test/node_modules/doqmentdb/node_modules/bluebird/js/main/promise.js:454:31)
    at Promise._settlePromiseAt (/home/agonbina/Playground/doqmentdb-test/node_modules/doqmentdb/node_modules/bluebird/js/main/promise.js:530:18)
    at Promise._settlePromises (/home/agonbina/Playground/doqmentdb-test/node_modules/doqmentdb/node_modules/bluebird/js/main/promise.js:646:14)
    at Async._drainQueue (/home/agonbina/Playground/doqmentdb-test/node_modules/doqmentdb/node_modules/bluebird/js/main/async.js:175:16)
    at Async._drainQueues (/home/agonbina/Playground/doqmentdb-test/node_modules/doqmentdb/node_modules/bluebird/js/main/async.js:185:10)
    at Async.drainQueues (/home/agonbina/Playground/doqmentdb-test/node_modules/doqmentdb/node_modules/bluebird/js/main/async.js:15:14)
    at process._tickCallback (node.js:442:13)
```

**Comment by a8m** [comment]
Thanks @agonbina , fixed via **v0.3.0**
Please let me know how it's goes.

**Comment by agonbina** [comment]
Great, that error is now gone. 

Now the other issue which I don't think is related to the library, is that if you run the interval every 100ms(sending 50 create requests to DocumentDB), running over 2 seconds a lot of(about 10-20% of the requests) end up throwing ```getaddrinfo ENOTFOUND``` errors. For example, out of 900 requests, 221 of them resulted in this same error in the first attempt. Any idea why this might be, or am I just sending way too many requests in such a small interval?
